hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/rollback/ListingBasedRollbackStrategy.java

```diff
@@ -57,20 +84,242 @@ public ListingBasedRollbackStrategy(HoodieTable table,
+              hoodieRollbackRequests.add(new HoodieRollbackRequest(
+                  partitionPath, EMPTY_STRING, EMPTY_STRING, filesToBeDeleted, Collections.emptyMap()));
+              break;
+            case APPEND_ROLLBACK_BLOCK:
+              HoodieCommitMetadata commitMetadata = HoodieCommitMetadata.fromBytes(
+                  table.getMetaClient().getCommitTimeline().getInstantDetails(instantToRollback).get(),
+                  HoodieCommitMetadata.class);
+
+              hoodieRollbackRequests.addAll(getRollbackRequestForAppend(partitionPath, instantToRollback, commitMetadata, table));
+              break;
+            default:
+              break;
+          }
+        }
+        return hoodieRollbackRequests.stream();
+      }, numPartitions);
+    } catch (Exception e) {
       LOG.error("Generating rollback requests failed for " + instantToRollback.getTimestamp(), e);
       throw new HoodieRollbackException("Generating rollback requests failed for " + instantToRollback.getTimestamp(), e);
     }
   }
+
+  private FileStatus[] getBaseFilesToBeDeleted(HoodieTableMetaClient metaClient, HoodieWriteConfig config,
+                                               String commit, String partitionPath, FileSystem fs) throws IOException {
+    LOG.info("Collecting files to be cleaned/rolledback up for path " + partitionPath + " and commit " + commit);
+    String basefileExtension = metaClient.getTableConfig().getBaseFileFormat().getFileExtension();
+    PathFilter filter = (path) -> {
+      if (path.toString().contains(basefileExtension)) {
+        String fileCommitTime = FSUtils.getCommitTime(path.getName());
+        return commit.equals(fileCommitTime);
+      }
+      return false;
+    };
+    return fs.listStatus(FSUtils.getPartitionPath(config.getBasePath(), partitionPath), filter);
+  }
+
+  private FileStatus[] getBaseAndLogFilesToBeDeleted(HoodieTableMetaClient metaClient, HoodieInstant instantToRollback, String partitionPath) throws IOException {
+    HoodieActiveTimeline activeTimeline = metaClient.getActiveTimeline();
+    HoodieTimeline timeline = activeTimeline.getCommitsTimeline().filterCompletedInstants();
+    Option<HoodieCommitMetadata> commitMetadataOptional = getHoodieCommitMetadata(timeline, instantToRollback);
+    SerializablePathFilter pathFilter = getSerializablePathFilter(metaClient, instantToRollback.getTimestamp());
+    Path[] filePaths;
+    if (commitMetadataOptional.isPresent() && instantToRollback.isCompleted()
+        && !WriteOperationType.UNKNOWN.equals(commitMetadataOptional.get().getOperationType())) {
+      filePaths = getFilesFromCommitMetadata(metaClient, commitMetadataOptional.get(), partitionPath);
+    } else {
+      filePaths = listFilesToBeDeleted(metaClient, partitionPath);
+    }
+
+    return metaClient.getFs().listStatus(filePaths, pathFilter);
+  }
+
+  private static Path[] listFilesToBeDeleted(HoodieTableMetaClient metaClient, String partitionPath) {
```

Comment:
If we're just accessing the base-path there's no need to pass whole `metaClient`, right?